crates/bevy_winit/src/state.rs

```diff
@@ -223,16 +223,22 @@ impl<T: Event> ApplicationHandler<T> for WinitAppRunnerState<T> {
                 );
             }
             WindowEvent::CloseRequested => self.winit_events.send(WindowCloseRequested { window }),
-            WindowEvent::KeyboardInput { ref event, .. } => {
-                if event.state.is_pressed() {
-                    if let Some(char) = &event.text {
-                        let char = char.clone();
-                        #[allow(deprecated)]
-                        self.winit_events.send(ReceivedCharacter { window, char });
+            WindowEvent::KeyboardInput {
+                ref event,
+                is_synthetic,
+                ..
+            } => {
+                if !(is_synthetic && event.state == ElementState::Pressed) {
```

Comment:
The check here and immediately below about whether the event state is pressed seem redundant to me.